contracts/validator-manager/PoSValidatorManager.sol

```diff
@@ -482,11 +432,15 @@ abstract contract PoSValidatorManager is
         if (delegator.owner != _msgSender()) {
             revert InvalidAddress();
         }
+        // Delegators may not initiate removal when the validator's state is pending.
+        if (validator.status == ValidatorStatus.PendingRemoved) {
```

Comment:
Could we make this a bit more defensive and check `if (validator.status != ValidatorStatus.Active)`?

Reply:
Ah nvm I see we handle the `Completed` case also. Can we move the check a single if/else if/else block though? 

ie.
```
if (status == active){
    ...
} else if (status == completed) {
    ...
} else {
   revert
}

Reply:
I refactored this a bunch in **https://github.com/ava-labs/teleporter/pull/573/commits/9dbb1faa8ebbf54013a7f92f876aa4bc3b3afa30**, let me know what you think.